crates/ra_assists/src/assists/auto_import.rs

```diff
@@ -36,6 +36,8 @@ pub(crate) fn auto_import<F: ImportsLocator>(
     if path_to_import_syntax.ancestors().find(|ancestor| ancestor.kind() == USE_ITEM).is_some() {
         return None;
     }
+    let name_to_import =
+        path_to_import_syntax.descendants().find(|child| child.kind() == NAME_REF)?;
```

Comment:
I see the pattern `.find(|child| child.kind() == NAME_REF)` is becomming pervasive.
Might you consider extracting this lambda into a function?
```rust
fn with_kind(kind: SyntaxKind) -> impl Fn(node: &SyntaxNode) -> bool {
     |node| node.kind() == kind
}
```
Or anything more appropriate from you point of view

Reply:
I'd write this as `.find_map(ast::NameRef::cast)`, just to type more things

Reply:
Hm, I'd even say that it's probably better to stick with the well-typed API:

```rust
fn first_component(path: ast::Path) -> ast::PathSegment {
    std::iter::successors(Some(path), |path| path.parent_path())
        .last().and_then(ast::Path::segment)
}
```

that's even more typing, but hopefully should be more rigid.

Reply:
bors r=me with the refactoring applied

Reply:
Erm, apparently Bors had had own plans on this PR :)

I'll create another one, thanks for the suggestions.

Reply:
It's because `bors r=me` is the same as `bors r+` with `me` as reviewer.

Reply:
yeah, it's that I actually changed my mind mid sentence as to whether this should be merged immediately or gated on refactoring :D

Reply:
https://github.com/rust-analyzer/rust-analyzer/pull/2994

I've decided to go with the `.find_map(ast::NameRef::cast)` since the other proposal uses `successors` in it while we also use `descendants` in other cases.

I've also used the opportunity to add one `;` to the inlay hints test snipped that bugged me, but was not big enough for a separate PR :)